src/class/msc/msc_device.h

```diff
@@ -164,6 +182,12 @@ ATTR_WEAK void tud_msc_read10_complete_cb(uint8_t lun);
 ATTR_WEAK void tud_msc_write10_complete_cb(uint8_t lun);
 ATTR_WEAK void tud_msc_scsi_complete_cb(uint8_t lun, uint8_t const scsi_cmd[16]);
 
+// Hook to make a mass storage device read-only.
+ATTR_WEAK bool tud_msc_is_writable_cb(uint8_t lun);

```

Comment:
Yeah, stack lack this feature, but it shouldn't be implemented as callback I think. It should be set by application using an class API by setting mode sense/sensekey response similar to tud_msc_set_sense() . It is OK to merge this now, but I will likely to change it later

Reply:
Why would you change it? This approach means tinyusb doesn't need to store extra state on its own. The compiler can optimize the call away if it never changes.

Reply:
since this is rather specific, not as generic as an usb stack should be. An sense/disk readiness state should be actively set by application to inform the stack.

Reply:
I am not sure either, most user is not familiar with MSC sense/sensekey etc.. Maybe leave it as you did for now, I check the scsi docs and find a better way